clang/lib/CIR/CodeGen/CIRGenItaniumCXXABI.cpp

```diff
@@ -2173,3 +2185,135 @@ void CIRGenItaniumCXXABI::buildThrow(CIRGenFunction &CGF,
   builder.create<mlir::cir::ThrowOp>(CGF.getLoc(E->getSourceRange()),
                                      exceptionPtr, typeInfo.getSymbol(), dtor);
 }
+
+static mlir::cir::FuncOp getBadCastFn(CIRGenFunction &CGF) {
+  // Prototype: void __cxa_bad_cast();
+  mlir::cir::FuncType FTy =
+      CGF.getBuilder().getFuncType({}, CGF.getBuilder().getVoidTy());
+  return CGF.CGM.getOrCreateRuntimeFunction(FTy, "__cxa_bad_cast");
+}
+
+void CIRGenItaniumCXXABI::buildBadCastCall(CIRGenFunction &CGF,
+                                           mlir::Location loc) {
+  CGF.buildRuntimeCall(loc, getBadCastFn(CGF));
+  // TODO(cir): mark the current insertion point as unreachable.
```

Comment:
Please also add one more unimplemented assert here (just like mentioned in the setCallingConv comment)

Reply:
Should we introduce a `cir.unreachable` operation in the future to be used in places like here?

Reply:
Updated.

Reply:
We should, I missed that, thanks for catching it. Can you add one more unimplemented assert here for unreachable?

Reply:
Updated.